src/main/java/overlap_use_case/TimetableModelUnpacker.java

```diff
@@ -0,0 +1,95 @@
+package overlap_use_case;
+
+import entities.*;
+import retrieve_timetable_use_case.*;
+
+import java.util.ArrayList;
+
+/** A class responsible for converting between Timetable Model classes (that is, including its internal
+ * model classes) back into their entity forms. "Unpacks the model class" into the entity.
+ */
+public class TimetableModelUnpacker {
+
+    public static Session unpackSessionModel(SessionModel sessionModel){
+
+        Session session = new Session(sessionModel.getSessionType());
+
+        for (CourseModel courseModel : sessionModel.getCourses().values()){
+            session.addCourse(unpackCourseModelAsCalendarCourse(courseModel));
+        }
+
+        return session;
+    }
+
+    public static Timetable unpackTimeTable(TimetableModel timetable){
+        ArrayList<TimetableCourse> courses = new ArrayList<>();
+        for (CourseModel course : timetable.getCourses() ){
+            courses.add(unpackCourseModelasTimetableCourse(course));
+        }
+
+        return new Timetable(courses, timetable.getSessionType());
+    }
+
+    // Since this class is specifically for unpacking Timetables, we make a TimetableCourse.
+    public static TimetableCourse unpackCourseModelasTimetableCourse(CourseModel course){
+
+        ArrayList<Section> sections = new ArrayList<>();
+        for (SectionModel sectionModel : course.getSections()){
+            sections.add(unpackSectionModel(sectionModel));
+        }
+
+        try {
+            return new TimetableCourse(course.getTitle(), sections, course.getCourseSession(),
+                    course.getCourseCode(), course.getBreadth());
+        } catch (InvalidSectionsException e){
+            throw new IllegalArgumentException("You tried to unpack a CourseModel that has invalid sections!");
+        }
+    }
+
+    // FIXME: I think this might be a code smell. Two nearly identical methods with one slightly changed constructor call.
```

Comment:
If you have a bunch of repeated code between unpackCourseModelAsCalendarCourse and unpackCourseModelasTimetableCourse, it may help to write a private helper method that completes the shared code (if the shared code is particularly long). Otherwise, this seems like the best we can do right now.

Reply:
Well, I pulled out the loop. But it's literally just unpackSection() over a list, and while it doesn't hurt to factor it out, it is just one loop.

The rest of the method is pretty much just a return statement, which can't be usefully simplified imo. 

Thanx tho! :3